runtime/lua/vim/lsp.lua

```diff
@@ -951,7 +951,7 @@ function lsp.start_client(config)
   ---second result. You can use this with `client.cancel_request(request_id)`
   ---to cancel the-request.
   ---@see |vim.lsp.buf_request()|
-  function client.request(method, params, handler, bufnr)
+  function client.request(method, params, handler, bufnr, on_cancel)
```

Comment:
it's not really "on cancel", it's "on_complete regardless of if cancelled. Need to rename